### ISSUE TITLE
Document the way the Kubernetes scheduler mode impacts Karpenter

### ISSUE THREAD

**Issue by stevehipwell** [kubernetes-sigs/karpenter#1228]
### Description

**What problem are you trying to solve?**
I have the following hypothesis.

> With the default Kubernetes scheduler configuration of `LeastAllocated`, Karpenter is statistically more likely to disrupt pods and/or require consolidation than if the scheduler was running in `MostAllocated` mode.

I'd like to see this tested and documented. I suspect that in some cases, especially where eviction on some pods is disabled, that this would provide a significant price reduction.

**How important is this feature to you?**
I think having this officially documented would help users understand how to get the most out of Karpenter and would add weight to requests to cloud providers to support customising the scheduler.

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by stevehipwell** [comment]
CC @jonathan-innis (this is what I spoke to you at KubeCon EU about)

**Comment by jonathan-innis** [comment]
> I'd like to see this tested and documented. I suspect that in some cases, especially where eviction on some pods is disabled, that this would provide a significant price reduction

Agreed. This sounds like a good feature to explore. Obviously, for us to give a recommendation for those that have direct control of their SchedulerConfiguration and the control plane, but also platforms like EKS should explore this alternative option to see if they get increased performance with Karpenter from this other setting.

I suspect that you are right about the performance, at least that we would be able to act more aggressively with consolidation since we'd have "less cluster churn" disrupting our current consolidation decision-making.

**Comment by jonathan-innis** [comment]
I think folks are recognizing that this is more of a problem here (and I'd agree). See: #1851

**Comment by stevehipwell** [comment]
@jonathan-innis the [Cloud-Agnostic Approach to Bin-Packing Pods in Managed Kubernetes in AWS, GCP and Azure](https://youtu.be/3podlIoxwyI?feature=shared&t=1378) talk at KubeCon EU 2024 covers having to run a custom scheduler for Karpenter to make use of `MostAllocated`.

**Comment by Jayanta1696** [comment]
I would like to work on it, if it's still open and needs work.

**Comment by VPC-byte** [comment]
I would like to work on this. My plan is to add a small docs section that explains how Kubernetes scheduler scoring mode can affect pod spreading/bin-packing and Karpenter consolidation, with a minimal local or KWOK reproduction if maintainers want measured evidence. Is that scope still useful?
